File: pset2/Caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

int main(int argc, string argv[]){

    //Check usage
    if(argc==2){
        int lenght = strlen(argv[1]);
        for(int i=0; i < lenght; i++){
            if(isdigit(argv[1][i]) == 0){
                printf("Usage: ./caesar key\n");
                    return 1;
            }
        }
    }else{
        printf("Usage: ./caesar key\n");
        return 1;
    }

    //Define variable
    string plainText=get_string("plaintext: ");
    int key = atoi(argv[1]);
    int lenght = strlen(plainText);
    char cipherText[lenght+1];

    //Caesar password
    for(int i=0;i<lenght;i++){
        if(isalpha(plainText[i]) != 0){
            if(islower(plainText[i]) != 0){
                cipherText[i]=(plainText[i]%97+key)%26+97;
            }else{
                cipherText[i]=(plainText[i]%65+key)%26+65;
            }
        }else{
            cipherText[i]=plainText[i];
        }
    }

    //Print cipher text
    printf("ciphertext: ");

    for(int i=0;i<lenght;i++)
        printf("%c",cipherText[i]);

    printf("\n");
}

File: pset6/Caesar.py
from sys import argv,exit

def main ():

    #Check usage
    if len(argv) !=2:
        print("Usage: python caesar.py k")
        exit(1)
    elif not argv[1].isnumeric():
        print("Usage: python caesar.py k")
        exit(1)
    elif int(argv[1]) <= 0:
        print("Usage: python caesar.py k")
        exit(1)
    else:
        #Get plain text
        plaintext = input("plaintext:")

        #Define cipher text
        ciphertext = []

        #Define k
        k = int(argv[1])
        # Get cipher text
        for i in range(len(plaintext)):
            if plaintext[i].islower():
                ciphertext.append(chr(((ord(plaintext[i])%97+k)%26+97)))
            elif plaintext[i].isupper():
                ciphertext.append(chr(((ord(plaintext[i])%65+k)%26+65)))
            else:
                ciphertext.append(plaintext[i])
        #Print cipher text
        print("ciphertext: ",end="")
        for i in range(len(ciphertext)):
            print(ciphertext[i],end="")
        print()

if __name__ == "__main__":
    main()